### PR TITLE
feat: add template for placeholder

### PR DESCRIPTION
**Pull request by SimonSoftwareSolution**
#1505 
#1555

### CODE REVIEW

**Comment by SimonSoftwareSolution** [comment]
Can someone help me with the CodeFactor Issue ? 
I've searched for an hour for a solution but couldnt find one...

**Comment by Shaper9** [comment]
Double check if the naming convention is right

**Comment by SimonSoftwareSolution** [comment]
@Shaper9 Its the same selector naming without the ng prefix in all the other examples as well.

**Comment by pavankjadda** [comment]
@SimonSoftwareSolution The placeholder stays even after selecting the item. See below

Screenshot 2024-07-25 at 4 07 22 PM

File: src/ng-select/lib/ng-select.component.html [hunk 1]
```diff
@@ -4,8 +4,13 @@
...
 	class="ng-select-container">
 	<div class="ng-value-container">
-		<div class="ng-placeholder">{{ placeholder }}</div>
-
+        <ng-template #defaultPlaceholderTemplate>
+            <div class="ng-placeholder">{{ placeholder }}</div>
+        </ng-template>
+        <ng-template
+			[ngTemplateOutlet]="placeholderTemplate || defaultPlaceholderTemplate">
+        </ng-template>
```
Comment (pavankjadda) [inline src/ng-select/lib/ng-select.component.html:12]: The placeholder stays even after selecting the item. See below

Screenshot 2024-07-25 at 4 07 22 PM

**Review by pavankjadda — CHANGES_REQUESTED** [review]

File: src/ng-select/lib/ng-select.component.html (hunk 1 above)
Comment (SimonSoftwareSolution) [reply src/ng-select/lib/ng-select.component.html:13]: Should be fixed, also removing the placeholder div in the DOM anymore, so people can work with :before Element

**Comment by pavankjadda** [comment]
@SimonSoftwareSolution Can you check failed test cases https://github.com/ng-select/ng-select/actions/runs/10250224834/job/28355771084?pr=2407#step:6:111?

**Review by pavankjadda — CHANGES_REQUESTED** [review]
Check failed test cases https://github.com/ng-select/ng-select/actions/runs/10250224834/job/28355771084?pr=2407#step:6:111

**Comment by SimonSoftwareSolution** [comment]
@pavankjadda Tests are green again.

**Comment by pavankjadda** [comment]
#2419 takes care of this with changes

**Pull request closed without merging**